app/js/scripts.js

```diff
@@ -134,148 +134,239 @@ function editReportById(reportId){
+    updateQueryStringParam('reportId', reportId);
   })
-  .fail(function(xhr){
-    console.log(xhr);
-  });
+    .fail(function (xhr) {
+      console.log(xhr);
+    });
 }
 
-function runExistingReportByUrl(){
+function loadReportById(reportId) {
+  reloadReportPane(false);
+  //Make the request to the server to get a saved report
+  var report = getRequestObject('/reports/id/' + reportId, null, 'GET');
+  report.done(function (reportObject) {
+    //Load the existing report view
+    $("#new-report-parent").hide();
+    console.log(reportObject.line_items);
+    addMultipleReportLineItems(reportObject.line_items);
+    //Fill in and show the fields to select
+    //Show the new report UI
+    changeReportView('computer', 'view');
+    $("#report-name-field").html(reportObject.name);
+    updateQueryStringParam('reportId', reportId);
+  })
+    .fail(function (xhr) {
+      console.log(xhr);
+    })
+}
+
+async function calculateReportResultCount() {
+  var reportObject = await asyncGetRequestObject('/reports/dashboard/', null, 'GET'); //gets all reports in mySQL
+  var outputObj = reportObject;
+  var reportCalls = [];
+  for (var i = 0; i < reportObject.length; i++) { //for each report
+    reportCalls = await asyncGetRequestObject('/reports/' + reportObject[i].id + '/count', null, 'GET');
+    outputObj[i].size = reportCalls.count;
+  }
+  return outputObj;
+}
+
+async function loadReportsWithDashboard() { 
+  data = await calculateReportResultCount()
+  return data;
+}
+
+function createChart(){
+  var ctx = document.getElementById('deviceCountChart').getContext('2d');
+    loadReportsWithDashboard().then(data => {
+        var chartData = []
+        var chartLabels = []
+        var chartColors = []
+        for (var i = 0; i < data.length; i++) {
+          chartData.push(data[i].size)
+          chartLabels.push(data[i].name)
+          chartColors.push(randomColor())
+        }
+        var deviceCountChart = new Chart(ctx, {
+          type: 'horizontalBar',
+          data: {
+              labels: chartLabels,
+              datasets: [{
+                label: 'Device Count',
+                data: chartData,
+                backgroundColor: chartColors,
+              }]
+          },
+          options: {
+              scales: {
+                xAxes: [{
+                    ticks: {
+                      beginAtZero: true
+                    }
+                }],
+              },
+              legend: {
+                display: false
+              },
+              animation: {
+                duration: 500,
+                easing: 'easeInQuad'
+              }
+          }
+        });
+    });
+
+    loadAllUsers().then(data => {
+      $('#userCountCard').show();
+      $('#userCard').text(data.length)
+    }).catch(() => {
+      $('#userCountCard').hide();
+    })
+}
+
+async function loadAllUsers() {
+  var users = await asyncGetRequestObject('/users/all', null, 'GET');
```

Comment:
could do this in one line `return await asyncGetRequestObject('/users/all', null, 'GET');`